runtime/lib/common/system_desc.cpp

```diff
@@ -227,7 +227,7 @@ static std::unique_ptr<::tt::runtime::SystemDesc> getCurrentSystemDescImpl(
 
     auto dramUnreservedEnd = calculateDRAMUnreservedEnd(device);
 
-    chipDescs.push_back(::tt::target::CreateChipDesc(
+    chipDescs.emplace_back(::tt::target::CreateChipDesc(
```

Comment:
remark: this is unrelated to my changes, but I'd expect `push_back` to be noticeable inferior to `emplace_back` for this sort of case.  I also see a similar pattern in a few other places.  Am I missing something?

Reply:
Don't have a strong opinion here, but for this case it should be the same performance wise, as emplace_back will be invoking the copy constructor anyways.

I think where we get performance gains is if we construct the object in-place without using the copy-constructor or we invoke the move constructor with emplace_back(std::move(...));

Reply:
Hmm, are you sure? I would expect an rvalue to be move-constructed here. https://stackoverflow.com/questions/21858187/why-does-vector-emplace-back-call-the-move-constructor

Reply:
(I guess I agree it's worse than actually constructing in-place, but I think it would be equivalent to the `std::move(...)` you mention)

Reply:
Oh nice catch, I didn't notice this was a function call, thought it was an object being constructed in-place `::tt::target::ChipDesc(...)`

Reply:
I would have expected an in-place object to also behave the same, since it'd be an r-value, but I made a quick test-case, and it seems like in both of these cases, `push_back` uses the move ctor as well, so (assuming my toy example is representative of this case), you're right and `push_back` is completely equivalent
```
  normal ctor
push_back an existing obj: 
  copy ctor
emplace_back an existing obj: 
  copy ctor
push_back a func result: 
  normal ctor
  move ctor
emplace_back a func result: 
  normal ctor
  move ctor
push_back a temp: 
  normal ctor
  move ctor
emplace_back a temp: 
  normal ctor
  move ctor
```